### PR TITLE
Introduce GetFileArg and use it where possible

### PR DESCRIPTION
**Pull request by prusnak**
Follow-up to https://github.com/bitcoin/bitcoin/pull/24265

~Marking as draft until https://github.com/bitcoin/bitcoin/pull/24265 is merged as this PR builds on top of that PR~

### CODE REVIEW

File: src/util/system.cpp [hunk 1]
```diff
@@ -399,6 +386,23 @@ std::optional<unsigned int> ArgsManager::GetArgFlags(const std::string& name) co
     return std::nullopt;
 }
 
+fs::path ArgsManager::GetFileArg(std::string pathlike_arg, const fs::path& pathDefault) const
```
Comment (kiminuo) [inline src/util/system.cpp:389]: `pathlike_arg` is in snake case notation, `pathDefault` is in camel case notation.

Comment (prusnak) [reply src/util/system.cpp:389]: Right. Left comment in the PR this PR builds on: https://github.com/bitcoin/bitcoin/pull/24265#pullrequestreview-874013646

Comment (prusnak) [reply src/util/system.cpp]: Addressed in c71437c93611094aa57eb7da0e166159ff128bdd

**Comment by ryanofsky** [comment]
Concept ACK, but I think PR description should be clearer. Also, I think it would be better to drop the first commit 42688109b38063902f4c2621db3e465a3c9b2218 here, and substitute commit 3206d1aec89a3d0d4c942d3d85b7e8cc8deb9881 from #24306. Main reason is that I think it is confusing to have separate `GetPathArg` and `GetFileArg` methods which are exactly the same except one strips trailing slashes and one doesn't. I don't think it's likely we will have a need for the one which preserves trailing slashes. The commit from the other PR also adds unit test coverage.

**Comment by prusnak** [comment]
Let's close this in favor of https://github.com/bitcoin/bitcoin/pull/24306

**Pull request closed without merging**